spyder/api/plugin_registration/mixins.py

```diff
@@ -38,15 +39,41 @@ def __init__(self):
         for method_name in dir(self):
             method = getattr(self, method_name, None)
             if hasattr(method, '_plugin_listen'):
-                info = method._plugin_listen
-                logger.debug(f'Method {method_name} is watching plugin {info}')
-                self._plugin_listeners[info] = method_name
+                plugin_listen = method._plugin_listen
+
+                # Check if plugin is listed among REQUIRES and OPTIONAL
+                if (
+                    self.NAME != Plugins.Layout and
```

Comment:
Could you add here a comment explaining the exception done with the Layout plugin?

Reply:
Good point, I expanded the comment above this line to explain that.